segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/MetadataStore.java

```diff
@@ -141,6 +144,10 @@ public void close() {
      * </ul>
      */
     CompletableFuture<Void> createSegment(String segmentName, SegmentType segmentType, Collection<AttributeUpdate> attributes, Duration timeout) {
+        if (segmentType.isTransientSegment() && !NameUtils.isTransientSegment(segmentName)) {
+            return Futures.failedFuture(new IllegalArgumentException("Invalid name '" + segmentName + "' for SegmentType: Transient"));
```

Comment:
tip 1: Use `String.format`
tip 2: don't hardcode `Transient`; use `segmentType.toString`(or if you use `String.format`, omit the `toString` as it will automatically invoke it.